tuxedo-core/src/verifier.rs

```diff
@@ -38,11 +38,43 @@ pub trait Verifier: Debug + Encode + Decode + Clone {
     /// The actual verification logic
     fn verify<R: Redeemer>(
         &self,
-        inputs: &[Input],
+        inputs: &[Output<R>],
         outputs: &[Output<R>],
     ) -> Result<TransactionPriority, Self::Error>;
 }
 
+// This blanket implementation makes it so that any type that chooses to
+// implement the Simple trait also implements the Powerful trait. This way
+// the executive can always just call the Powerful trait.
+impl<T: SimpleVerifier> Verifier for T {
+
+    // When using the blanket implementation, we directly return a UtxoError here
+    // So that we have access to the `RedeemerError` variant
+    type Error = <T as SimpleVerifier>::Error;
+
+    fn verify<R: Redeemer>(
+        &self,
+        inputs: &[Output<R>],
+        outputs: &[Output<R>],
+    ) -> Result<TransactionPriority, Self::Error> {
+
+        // Extract the input data
+        let input_data: Vec<DynamicallyTypedData> = inputs
+            .iter()
+            .map(|o| o.payload.clone())
+            .collect();
+
+        // Extract the output data
+        let output_data: Vec<DynamicallyTypedData> = outputs
+            .iter()
+            .map(|o| o.payload.clone())
+            .collect();
+
+        // Call the simple verifier
+        SimpleVerifier::verify(self, &input_data, &output_data)
+    }
```

Comment:
This is the main change. A simple addition that makes the macro unnecessary.

Reply:
Ahh ok you can do this perfect!